tests/angularModuleSpec.js

```diff
@@ -397,3 +397,60 @@ describe('TaskCtl', function () {
         expect(Logging.level).toEqual(2);
     });
 });
+
+describe('StateCtrl', function () {
+    var $httpBackend, adalServiceProvider, rootScope, state, location, $templateCache, $stateParams;
+
+    //mock Application to allow us to inject our own dependencies
+    beforeEach(angular.mock.module('StateApplication'));
+
+    //mock the controller for the same reason and include $scope and $controller
+    beforeEach(angular.mock.inject(function (_adalAuthenticationService_, _$rootScope_, _$httpBackend_, _$state_, _$location_, _$templateCache_, _$stateParams_) {
+        adalServiceProvider = _adalAuthenticationService_;
+        rootScope = _$rootScope_;
+        $httpBackend = _$httpBackend_;
+        state = _$state_;
+        location = _$location_;
+        $templateCache = _$templateCache_;
+        $stateParams = _$stateParams_;
+        $httpBackend.expectGET('settings.html').respond(200);
+        $httpBackend.expectGET('profile.html').respond(200);
+        $httpBackend.expectGET('name.html').respond(200);
+        $httpBackend.expectGET('account.html').respond(200);
+        $templateCache.put('profile.html', '');
+        $templateCache.put('settings.html', '');
+        $templateCache.put('account.html', '');
+        $templateCache.put('name.html', '');
+        adalServiceProvider.config.anonymousEndpoints = [];
+    }));
+
+    it('checks if anonymous endpoints are populated on statechange event if states are nested and separated by .', function () {
+        rootScope.$on('$stateChangeSuccess', function (event, toState) {
+            state = toState;
```

Comment:
That is a reference to $state, we should not update that. use a local variable to assign "toState" to.